src/wallet/asyncrpcoperation_sendmany.cpp

```diff
@@ -0,0 +1,945 @@
+ */
+void AsyncRPCOperation_sendmany::sign_send_raw_transaction(Object obj)
+{   
+    // Sign the raw transaction
+    Value rawtxnValue = find_value(obj, "rawtxn");
+    if (rawtxnValue.is_null()) {
+        throw JSONRPCError(RPC_WALLET_ERROR, "Missing hex data for raw transaction");
+    }
+    std::string rawtxn = rawtxnValue.get_str();
+
+    Value signResultValue = signrawtransaction({Value(rawtxn)}, false);
+    Object signResultObject = signResultValue.get_obj();
+    Value completeValue = find_value(signResultObject, "complete");
+    bool complete = completeValue.get_bool();
+    if (!complete) {
+        // TODO: Maybe get "errors" and print array vErrors into a string
+        throw JSONRPCError(RPC_WALLET_ENCRYPTION_FAILED, "Failed to sign transaction");
+    }
+
+    Value hexValue = find_value(signResultObject, "hex");
+    if (hexValue.is_null()) {
+        throw JSONRPCError(RPC_WALLET_ERROR, "Missing hex data for signed transaction");
+    }
+    std::string signedtxn = hexValue.get_str();
+
+    // Send the signed transaction
+    if (!testmode) {
+        Value sendResultValue = sendrawtransaction({Value(signedtxn)}, false);
+        if (sendResultValue.is_null()) {
+            throw JSONRPCError(RPC_WALLET_ERROR, "Send raw transaction did not return an error or a txid.");
+        }
+
+        std::string txid = sendResultValue.get_str();
+
+        Object o;
+        o.push_back(Pair("txid", txid));
+        //o.push_back(Pair("hex", signedtxn));
+        set_result(Value(o));
+    } else {
+        // Test mode does not send the transaction to the network.
+
+        CDataStream stream(ParseHex(signedtxn), SER_NETWORK, PROTOCOL_VERSION);
+        CTransaction tx;
+        stream >> tx;
+
+        Object o;
+        o.push_back(Pair("test", 1));
+        o.push_back(Pair("txid", tx.GetTxid().ToString()));
+        o.push_back(Pair("hex", signedtxn));
+        set_result(Value(o));
+    }
+}
+
+
+bool AsyncRPCOperation_sendmany::find_utxos() {
+    set<CBitcoinAddress> setAddress = {fromtaddr_};
+    vector<COutput> vecOutputs;
+
+    LOCK2(cs_main, pwalletMain->cs_wallet);
+
+    pwalletMain->AvailableCoins(vecOutputs, false, NULL, true);
+
+    BOOST_FOREACH(const COutput& out, vecOutputs) {
+        if (out.nDepth < mindepth_) {
+            continue;
+        }
+
+        if (setAddress.size()) {
+            CTxDestination address;
+            if (!ExtractDestination(out.tx->vout[out.i].scriptPubKey, address)) {
+                continue;
+            }
+
+            if (!setAddress.count(address)) {
+                continue;
+            }
+        }
+
+        // TODO: Also examine out.fSpendable ?
+        CAmount nValue = out.tx->vout[out.i].nValue;
+        SendManyInputUTXO utxo(out.tx->GetTxid(), out.i, nValue);
+        t_inputs_.push_back(utxo);
+    }
+    return t_inputs_.size() > 0;
+}
+
+
+bool AsyncRPCOperation_sendmany::find_unspent_notes() {
```

Comment:
I'll be abstracting the contents of this method into the wallet for reuse in #1201, so that `find_unspent_notes()` is functionally identical to `find_utxos()`.